windows/src/engine/keyman32/hookutils.cpp

```diff
@@ -66,23 +65,26 @@ DWORD ExceptionMessage(LPSTR Proc, LPEXCEPTION_POINTERS ep)
     }
   }
 
-  if(!ep || !ep->ExceptionRecord) 
-  {
-    SendDebugMessageFormat(0, sdmGlobal, 0, "CAUGHT UNKNOWN EXCEPTION");
+  if(!ep || !ep->ExceptionRecord) {
+    OutputThreadDebugString("CAUGHT UNKNOWN EXCEPTION");
     return EXCEPTION_CONTINUE_SEARCH;
   }
   LPEXCEPTION_RECORD er = ep->ExceptionRecord;
   
-  while(er != NULL)
-  {
-    if(er->ExceptionCode == EXCEPTION_ACCESS_VIOLATION)
-    	SendDebugMessageFormat(0, sdmGlobal, 0, "CAUGHT EXCEPTION %d (ACCESS VIOLATION) IN %s AT %x; attempted to %s %x",
-    	  er->ExceptionCode, Proc, er->ExceptionAddress,
-    	  er->ExceptionInformation[0] == 0 ? "read from" : "write to",
-    	  er->ExceptionInformation[1]);
-    else
-  	  SendDebugMessageFormat(0, sdmGlobal, 0, "CAUGHT EXCEPTION %d IN %s AT %x",
-    	  er->ExceptionCode, Proc, er->ExceptionAddress);
+  while(er != NULL) {
+    if (er->ExceptionCode == EXCEPTION_ACCESS_VIOLATION) {
+      OutputThreadDebugString("EXCEPTION_ACCESS_VIOLATION in ");
+      OutputThreadDebugString(Proc);
+      SendDebugMessageFormat(0, sdmGlobal, 0, "CAUGHT EXCEPTION %d (ACCESS VIOLATION) IN %s AT %x; attempted to %s %x",
```

Comment:
I assume this (and the one in the following `else` block) weren't possible to replace like the others?

(in reference to https://github.com/keymanapp/keyman/pull/2297/files#r342938825)

Reply:
They add more information so I left them ... which may not be the best idea.